src/ir/module.h

```diff
@@ -62,7 +68,12 @@ class Operation {
     return visitor.Visit(*this);
   }
 
+  friend class ProtoToIR;
+
  private:
+  // Set the block to which this operation belongs. This is private so that only
+  // the Block can set it via its `AddOperation` function.
+  void set_parent_block(const Block& block) { parent_ = &block; }
```

Comment:
OK for now as this is private. However, @harsha-mandadi is adding a similar public method in one of her other PRs.

Reply:
Okay, depending on the merge order I can remove it :)